Algorithm/QCAlgorithm.Indicators.cs

```diff
@@ -1738,8 +1739,13 @@ public SubscriptionDataConfig GetSubscription(Symbol symbol, TickType? tickType)
                 // deterministic ordering is required here
                 var subscriptions = SubscriptionManager.Subscriptions.OrderBy(x => x.TickType);
 
+                if (tickType == null)
+                {
+                    tickType = LeanData.GetCommonTickType(symbol.SecurityType);
+                }
+
                 // find our subscription to this symbol
-                subscription = subscriptions.FirstOrDefault(x => x.Symbol == symbol && (tickType == null || tickType == x.TickType));
+                subscription = subscriptions.SingleOrDefault(x => x.Symbol == symbol && x.TickType == tickType);
```

Comment:
I'm not sure this should be Single? Can we guarantee there won't be other subscriptions for that symbol and tick type?
Wondering why change previous behavior if `tickType == null`?

Reply:
>I'm not sure this should be Single? Can we guarantee there won't be other subscriptions for that symbol and tick type?
To my best understand, the only case where we can have more than one subscription for a given symbol and tick type is if we add a symbol with multiple resolutions. 

But that case is not implemented in the actual master; there is no definition of which resolution it should return. 

Any thoughts?

> Wondering why change previous behavior if tickType == null?
After asking @AlexCatarino, he explained that in cases where the security has both tick types (crypto and soon equities),  the expected subscription should be the one with the _preferred_ tick type.

I added a test that fails in actual master for the crypto case. 
https://github.com/QuantConnect/Lean/blob/81f201e7dfa954f2081393551a8a73c8afcd0ee2/Tests/Algorithm/AlgorithmResolveConsolidatorTests.cs#L31

Reply:
True, but using `SingleOrDefault` seems like a breaking change compared to `master`, since it will throw, I'd be inclined to respect current behavior and use `FirstOrDefault` instead.

Reply:
Good point!